app/models/placements/term.rb

```diff
@@ -25,6 +25,6 @@ class Placements::Term < ApplicationRecord
 
   validates :name, presence: true, inclusion: { in: VALID_NAMES }
 
-  has_many :placement_windows, class_name: "Placements::PlacementWindow"
+  has_many :placement_windows, class_name: "Placements::PlacementWindow", dependent: :destroy
```

Comment:
How about using `dependent: :restrict_with_exception` here?

**We should never delete terms.** But if we ever accidentally do (e.g. somebody types a dodgy Rails console command), this would give an extra layer of protection so we don't inadvertently cascade delete a large chunk of placement records.

I like the built-in protection this offers. So I think it'd be sensible to make use of it here.

```suggestion
  has_many :placement_windows, class_name: "Placements::PlacementWindow", dependent: :restrict_with_exception
```